### ISSUE TITLE
AuditLogEntry channel_id TypeError

### ISSUE THREAD

**Issue by HexCodeFFF** [Rapptz/discord.py#9392]
### Summary

Regularly my bot will crash with a TypeError concerning the AuditLogEntry channel_id being None

### Reproduction Steps

My bot is in like 30k servers and it just happens on its own

### Expected Results

The bot to continue functioning as normal

### Actual Results

```Traceback (most recent call last):
  File "/mediaforge/src/main.py", line 194, in <module>
    bot.run(config.bot_token, log_handler=None)
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/client.py", line 860, in run
    asyncio.run(runner())
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 650, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/client.py", line 849, in runner
    await self.start(token, reconnect=reconnect)
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/client.py", line 778, in start
    await self.connect(reconnect=reconnect)
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/shard.py", line 464, in connect
    raise item.error
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/shard.py", line 164, in worker
    await self.ws.poll_event()
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/gateway.py", line 623, in poll_event
    await self.received_message(msg.data)
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/gateway.py", line 573, in received_message
    func(data)
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/state.py", line 1106, in parse_guild_audit_log_entry_create
    entry = AuditLogEntry(
            ^^^^^^^^^^^^^^
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/audit_logs.py", line 592, in __init__
    self._from_data(data)
  File "/root/.cache/pypoetry/virtualenvs/mediaforge-HXFskkj--py3.11/lib/python3.11/site-packages/discord/audit_logs.py", line 647, in _from_data
    channel_id = int(extra['channel_id'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

### Intents

discord.Intents.all() minus intents.presences minus intents.members

### System Information

- Python v3.11.0-final
- discord.py v2.2.2-final
- aiohttp v3.8.4
- system info: Linux 5.15.0-69-generic #76-Ubuntu SMP Fri Mar 17 17:19:29 UTC 2023

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

**Comment by AbstractUmbra** [comment]
This was resolved in 34a434b2afb605fb3cb7884608394c258b6ef962, if it is urgent then update and pin to this commit until a release is made!

**Comment by Rapptz** [comment]
Update to v2.2.3
